shell/src/main/java/tachyon/shell/TFsShell.java

```diff
@@ -166,73 +152,67 @@ private int loadPath(TachyonFS tachyonClient, TachyonURI filePath) throws IOExce
    * fail if the path given already exists in the filesystem.
    *
    * @param argv [] Array of arguments given by the user's input from the terminal
-   * @return 0 if command is successful, -1 if an error occurred.
    * @throws IOException
    */
-  public int copyFromLocal(String[] argv) throws IOException {
+  public void copyFromLocal(String[] argv) throws IOException {
     String srcPath = argv[1];
     TachyonURI dstPath = new TachyonURI(argv[2]);
     File src = new File(srcPath);
     if (!src.exists()) {
-      System.out.println("Local path " + srcPath + " does not exist.");
-      return -1;
+      throw new IOException("Local path " + srcPath + " does not exist.");
     }
     TachyonFS tachyonClient = createFS(dstPath);
-    int ret = copyPath(src, tachyonClient, dstPath);
-    if (ret == 0) {
-      System.out.println("Copied " + srcPath + " to " + dstPath);
-    }
-    return ret;
+    copyPath(src, tachyonClient, dstPath);
+    System.out.println("Copied " + srcPath + " to " + dstPath);
   }
 
-  private int copyPath(File src, TachyonFS tachyonClient, TachyonURI dstPath) throws IOException {
+  private void copyPath(File src, TachyonFS tachyonClient, TachyonURI dstPath) throws IOException {
     if (!src.isDirectory()) {
       TachyonFile tFile = tachyonClient.getFile(dstPath);
       if (tFile != null && tFile.isDirectory()) {
         dstPath = dstPath.join(src.getName());
       }
       int fileId = tachyonClient.createFile(dstPath);
-      if (fileId == -1) {
-        return -1;
-      }
-      tFile = tachyonClient.getFile(fileId);
-      Closer closer = Closer.create();
       try {
-        WriteType writeType =
-            mTachyonConf.getEnum(Constants.USER_DEFAULT_WRITE_TYPE, WriteType.CACHE_THROUGH);
-        OutStream os = closer.register(tFile.getOutStream(writeType));
-        FileInputStream in = closer.register(new FileInputStream(src));
-        FileChannel channel = closer.register(in.getChannel());
-        ByteBuffer buf = ByteBuffer.allocate(8 * Constants.MB);
-        while (channel.read(buf) != -1) {
-          buf.flip();
-          os.write(buf.array(), 0, buf.limit());
+        tFile = tachyonClient.getFile(fileId);
+        Closer closer = Closer.create();
```

Comment:
Feels better to put this `Closer` in the outside `try`, and remove the next `try`, put the next `finally` after the following `catch(IOException e)`, then we only have one level of `try {...} catch (...) {...} finally {...}`.